src/components/molecules/OakModalCenter/OakModalCenter.tsx

```diff
@@ -0,0 +1,257 @@
+  disableEscapeKey = false,
+  hideCloseButton = false,
+  modalFlexProps,
+  backdropFlexProps,
+  footerSlot,
+}: OakModalCenterProps) => {
+  const [scrollBorders, setScrollBorders] = useState({
+    top: false,
+    bottom: false,
+  });
+  const transitionRef = useRef<HTMLDivElement>(null);
+  const scrollBoxRef = useRef<HTMLDivElement>(null);
+
+  const checkForScroll = useCallback(() => {
+    // To show some borders when the content is scrollable
+    const scrollBoxElem = scrollBoxRef.current;
+    if (!scrollBoxElem) {
+      return;
+    }
+
+    setScrollBorders({
+      top: scrollBoxElem.scrollTop > 0,
+      bottom: scrollBoxElem.scrollHeight > scrollBoxElem.clientHeight,
+    });
+  }, []);
+
+  useLayoutEffect(() => {
+    checkForScroll();
+    scrollBoxRef.current?.addEventListener("scroll", checkForScroll);
+    return () => {
+      scrollBoxRef.current?.removeEventListener("scroll", checkForScroll);
+    };
+  }, [isOpen]);
+
+  useEffect(() => {
+    window.addEventListener("resize", checkForScroll);
+    return () => {
+      window.removeEventListener("resize", checkForScroll);
+    };
+  });
+
+  return createPortal(
+    <Transition
+      in={isOpen}
+      nodeRef={transitionRef}
+      addEndListener={(done) => {
+        transitionRef.current?.addEventListener("transitionend", done);
+      }}
+      timeout={0}
+      mountOnEnter
+      unmountOnExit
+    >
+      {(state) => (
+        <FadeInFlex
+          ref={transitionRef}
+          $state={state}
+          $position="fixed"
+          $inset="all-spacing-0"
+          $justifyContent="center"
+          $alignItems="center"
+          $zIndex="modal-dialog"
+        >
+          <OakBox
+            $position="fixed"
+            $inset="all-spacing-0"
+            $zIndex="behind"
+            $background="blackSemiTransparent"
+            style={{ backdropFilter: `blur(3px)` }}
+            data-testid="backdrop"
+            {...backdropFlexProps}
+          />
+          <OakFlex
+            $alignItems="center"
+            $justifyContent="center"
+            $maxWidth="all-spacing-23"
+            $width="100%"
+            $pa="inner-padding-l"
+          >
+            <FocusOnBox
+              onEscapeKey={() => !disableEscapeKey && onClose()}
+              onClickOutside={() => !disableBackdropClick && onClose()}
+              returnFocus
+              autoFocus
+              $width="100%"
+            >
+              <OakFlex
+                $flexDirection="column"
+                $background="white"
+                $borderRadius="border-radius-l"
+                $ba="border-solid-xl"
+                $borderColor="border-decorative1-stronger"
+                $width="100%"
+                $position="relative"
+                role="dialog"
+                style={{
+                  maxHeight: `calc(100vh - ${parseSpacing(
+                    "inner-padding-xl5",
+                  )} - ${parseSpacing("inner-padding-xl5")})`,
```

Comment:
Makes the dialogs max height the viewport minus 2 lots of padding (top and bottom).

Reply:
there is a $maxHeight prop with a set of heights; but maybe this is the only way to get a responsive height based on viewport.